lib/sisimai/bite/email/apachejames.rb

```diff
@@ -0,0 +1,169 @@
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['message-id'] && mhead['message-id'] =~ Re0[:'message-id']
+        match += 1 if mhead['received'].find { |a| a =~ Re0[:received] }
+        return if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        diagnostic = ''     # (String) Alternative diagnostic message
+        subjecttxt = nil    # (String) Alternative Subject text
+        gotmessage = -1     # (Integer) Flag for error message
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # Message details:
+            #   Subject: Nyaaan
+            #   Sent date: Thu Apr 29 01:20:50 JST 2015
+            #   MAIL FROM: shironeko@example.jp
+            #   RCPT TO: kijitora@example.org
+            #   From: Neko <shironeko@example.jp>
+            #   To: kijitora@example.org
+            #   Size (in bytes): 1024
+            #   Number of lines: 64
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->